### PR TITLE
implement prependers

### PR DESCRIPTION
**Pull request by itsmontoya**

### CODE REVIEW

File: hooks.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+func NewAuthBearerHook(apiKey string) func() Opts {
+	// Set authorization header
+	authorization := Header{
+		Key: "Authorization",
+		Val: fmt.Sprintf("Bearer %s", apiKey),
+	}
+
+	return func() (o Opts) {
+		// Create new headers using authorization header
+		headers := NewHeaders(authorization)
```
Comment (dhalman) [inline hooks.go:15]: why not just append new headers? i thought you disliked ducktype

File: requester.go [hunk 2]
```diff
@@ -115,6 +125,11 @@ func (r *Requester) newRequest(ctx context.Context, method, path string, body []
 }
 
 func (r *Requester) setOpts(req *http.Request, opts Opts) (err error) {
+	if r.prepender != nil {
+		// Prepender exists, prepend prepender opts to opts list
```
Comment (dhalman) [inline requester.go:129]: prepend prepender

Comment (itsmontoya) [reply requester.go:129]: Bahahah, great catch

**Comment by russiansmack** [comment]
@itsmontoya You can pass an Opt with the request of type Header and change your authorization header.. none of this is prepander prepending is needed that way... :P

File: hooks.go (hunk 1 above)
Comment (itsmontoya) [reply hooks.go]: I usually dislike ducktype, except in situations where we are procuring a value from a single-return func. In this case, adding a variable name provides a little more context than directly returning the value. Though, directly returning the value would also be perfectly acceptable.

File: requester.go (hunk 2 above)
Comment (itsmontoya) [reply requester.go]: Fixed!

**Pull request merged**